File: data/CJBench/Java/DeepseekCommented/HeapElement.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    // The key value associated with this object.
    private final double key;
    // Additional information associated with this object.
    private final Object additionalInfo;

    // Constructor that initializes both key and additionalInfo.
    public Main(double key, Object info) {
        this.key = key;
        this.additionalInfo = info;
    }

    // Constructor that initializes only the key, with additionalInfo set to null.
    public Main(double key) {
        this.key = key;
        this.additionalInfo = null;
    }

    // Getter method for retrieving the additional information.
    public Object getInfo() {
        return additionalInfo;
    }

    // Getter method for retrieving the key.
    public double getKey() {
        return key;
    }

    // Provides a string representation of the object, including the key and additional information if available.
    @Override
    public String toString() {
        return "Key: " + key + " - " + (additionalInfo != null ? additionalInfo.toString() : "No additional info");
    }

    // Compares this object with another for equality based on key and additionalInfo.
    @Override
    public boolean equals(Object o) {
        if (o instanceof Main otherMain) {
            return this.key == otherMain.key
                && (this.additionalInfo != null ? this.additionalInfo.equals(otherMain.additionalInfo) : otherMain.additionalInfo == null);
        }
        return false;
    }

    // Generates a hash code for this object based on key and additionalInfo.
    @Override
    public int hashCode() {
        int result = 31 * (int) key;
        result += (additionalInfo != null) ? additionalInfo.hashCode() : 0;
        return result;
    }

    // Main method to run the command-line interface for managing Main objects.
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        ArrayList<Main> elements = new ArrayList<>();
        while (scanner.hasNextLine()) {
            String command = scanner.nextLine().trim();
            if (command.isEmpty()) {
                break;
            }
            String[] parts = command.split(" ");
            String operation = parts[0];
            try {
                switch (operation) {
                    case "create": 
                        double key = Double.parseDouble(parts[1]);
                        String info = parts.length > 2 ? parts[2] : null;
                        Main element = info != null ? new Main(key, info) : new Main(key);
                        elements.add(element);
                        System.out.println("Created: " + element);
                        break;
                    case "getKey": 
                        int index = Integer.parseInt(parts[1]);
                        if (index < 0 || index >= elements.size()) {
                            System.out.println("Error: Index " + index + " is out of bounds. Valid range: 0 to " + (elements.size() - 1));
                        } else {
                            System.out.println("Key: " + elements.get(index).getKey());
                        }
                        break;
                    case "getInfo": 
                        index = Integer.parseInt(parts[1]);
                        if (index < 0 || index >= elements.size()) {
                            System.out.println("Error: Index " + index + " is out of bounds. Valid range: 0 to " + (elements.size() - 1));
                        } else {
                            System.out.println("Info: " + elements.get(index).getInfo());
                        }
                        break;
                    case "print": 
                        for (Main el : elements) {
                            System.out.println(el);
                        }
                        break;
                    default:
                        System.out.println("Unknown operation: " + operation);
                }
            } catch (Exception e) {
                System.out.println("Error: " + e.getMessage());
            }
        }
        scanner.close();
    }
}

File: translations/CJBench/J2P/Uncommented/gpt-4o/Code/HeapElement.py
class Main:
    def __init__(self, key, info=None):
        self.key = key
        self.additional_info = info

    def get_info(self):
        return self.additional_info

    def get_key(self):
        return self.key

    def __str__(self):
        return f"Key: {self.key} - {self.additional_info if self.additional_info is not None else 'No additional info'}"

    def __eq__(self, other):
        if isinstance(other, Main):
            return (self.key == other.key and
                    (self.additional_info == other.additional_info if self.additional_info is not None else other.additional_info is None))
        return False

    def __hash__(self):
        result = 31 * int(self.key)
        result += hash(self.additional_info) if self.additional_info is not None else 0
        return result


def main():
    elements = []
    while True:
        command = input().strip()
        if not command:
            break
        parts = command.split(" ")
        operation = parts[0]
        try:
            if operation == "create":
                key = float(parts[1])
                info = parts[2] if len(parts) > 2 else None
                element = Main(key, info) if info is not None else Main(key)
                elements.append(element)
                print(f"Created: {element}")
            elif operation == "getKey":
                index = int(parts[1])
                if index < 0 or index >= len(elements):
                    print(f"Error: Index {index} is out of bounds. Valid range: 0 to {len(elements) - 1}")
                else:
                    print(f"Key: {elements[index].get_key()}")
            elif operation == "getInfo":
                index = int(parts[1])
                if index < 0 or index >= len(elements):
                    print(f"Error: Index {index} is out of bounds. Valid range: 0 to {len(elements) - 1}")
                else:
                    print(f"Info: {elements[index].get_info()}")
            elif operation == "print":
                for el in elements:
                    print(el)
            else:
                print(f"Unknown operation: {operation}")
        except Exception as e:
            print(f"Error: {e}")

if __name__ == "__main__":
    main()

# <END-OF-CODE>
